src/main/kotlin/Interfaces.kt

```diff
@@ -58,9 +58,9 @@ open class Car(
 
     override val topSpeed: Double = 100.0
 
-    override fun fillTank(station: ShellStation) {
-        station.acceptPayment(cashAvailable)
-        station.pumpGas(cashAvailable)
+    override fun fillTank(station: ServiceStation) {
+        (station as ShellStation).acceptPayment(cashAvailable)
```

Comment:
To handle the changed to a `ServiceStatation` parameter we have to cast the `station` as a `ShellStation` type to access the `acceptPayment` and `pumpGas` methods, which are on the `ShellStation`. Maybe not the best architecture, but the advantage is that you get to see how casts work.